File: LogicAndTrick.WikiCodeParser/HtmlHelper.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Web;

namespace LogicAndTrick.WikiCodeParser
{
    internal static class HtmlHelper
    {
        public static string Encode(string text)
        {
            return HttpUtility.HtmlEncode(text);
        }

        public static string UrlEncode(string urlPart)
        {
            return HttpUtility.UrlEncode(urlPart);
        }

        public static string AttributeEncode(string attributeText)
        {
            return HttpUtility.HtmlAttributeEncode(attributeText)
                .Replace(">", "&gt;")
                .Replace("'", "&#39;");
        }
    }
}

File: javascript/src/HtmlHelper.ts

function escapeEmoji(str : string) {
    return str.replace(/\p{Emoji_Presentation}/ugm, s => '&#' +s.codePointAt(0) + ';');
}

export class HtmlHelper {
    public static Encode(text: string): string {
        text = text.replace(/&/g, '&amp;')
            .replace(/</g, '&lt;')
            .replace(/>/g, '&gt;')
            .replace(/"/g, '&quot;')
            .replace(/'/g, '&#39;');
        return escapeEmoji(text);
    }

    public static UrlEncode(text: string): string {
        return encodeURI(text);
    }

    public static AttributeEncode(text: string): string {
        text = text.replace(/&/g, '&amp;')
            .replace(/</g, '&lt;')
            .replace(/>/g, '&gt;')
            .replace(/"/g, '&quot;')
            .replace(/'/g, '&#39;');
        return escapeEmoji(text);
    }
}
